packages/graphql/lib/schema-builder/factories/object-type-definition.factory.ts

```diff
@@ -189,18 +197,43 @@ export class ObjectTypeDefinitionFactory {
     TArgs = { [argName: string]: any },
     TOutput = any,
   >(field: PropertyMetadata, options: BuildSchemaOptions) {
-    const rootFieldResolver = (root: object) => {
+    const typeResolver = TypeMetadataStorage.getResolverMetadataFor(field);
+
+    const rootFieldResolver = async (...gqlArgs: any[]) => {
+      const root = this.gqlParamsFactory.exchangeKeyForValue(
+        GqlParamtype.ROOT,
+        undefined,
+        gqlArgs,
+      )
       const value = root[field.name];
+      if(typeResolver) {
+        const gqlContext = this.gqlParamsFactory.exchangeKeyForValue(
+          GqlParamtype.CONTEXT,
+          undefined,
+          gqlArgs,
+        )
+        const contextId = ContextIdFactory.getByRequest(gqlContext);
+        const instance = await this.moduleRef.resolve(typeResolver.target, contextId, { strict: false, });
+        if(instance) {
+          const annotatedMethod = this.metadataScanner.getAllMethodNames(instance).find(method => Reflect.hasMetadata(RESOLVE_OBJECT_TYPE_METADATA, instance[method]))
+          if(annotatedMethod) {
+            return instance[annotatedMethod](...gqlArgs);
+          }
```

Comment:
This ultimately does exactly what I want besides two things:
 - I don't know if it's possible to make this request scope without reaching into module ref.  I didn't quite follow what's going on in the resolver-explorer.service which seems to do something really similar to what I want to do.
 - I don't know how to get the graphql param decorators to work right. I think I need the `externalContextCreator` to do that, but I couldn't figure out the right set of params to get that working

Reply:
I think it would behoove this PR, if you would describe in detail what it is you want to accomplish with the PR i.e. fill in the form correctly in the PR's OP. I highly doubt anyone will take it seriously otherwise. 

Scott

Reply:
@smolinari , https://github.com/nestjs/graphql/issues/3023

Reply:
Forgive me for not having bidirectional linking here.  The main discussion existed in teh original issue, and over in discord.  This PR was opened as a draft and mentioned in those two locations, but not back linked here.

Reply:
Sorry, maybe I'm not understanding.  What exactly is it that I'm supposed to do here?  I've opened a draft PR and said that it's a draft for the purpose of illustrating a possible, but likely incomplete solution to a problem.  This is not an "open" PR.  

Is that not allowed per community guidelines?  I assumed a draft would be generally ignored and would be okay as a way to show my intent to people who were watching the issue that I'd raised.